{"version": 2, "width": 140, "height": 20, "timestamp": 1705969014, "idle_time_limit": 2.0, "env": {"SHELL": "/bin/zsh", "TERM": null}, "title": "step-19-SH"}
[0.01308, "o", "\u001b[0;97m"]
[0.016804, "o", "kafka-json"]
[0.2091, "o", "-schema-co"]
[0.309076, "o", "nsole-cons"]
[0.40913, "o", "umer \\\r\n   "]
[0.499826, "o", " --bootstr"]
[0.600287, "o", "ap-server "]
[0.698591, "o", "localhost:"]
[0.798122, "o", "6969 \\\r\n   "]
[0.898075, "o", " --consume"]
[0.997385, "o", "r.config t"]
[1.094252, "o", "eamA-sa.pr"]
[1.192712, "o", "operties \\"]
[1.29003, "o", "\r\n    --top"]
[1.389927, "o", "ic topic-j"]
[1.488831, "o", "son-schema"]
[1.581536, "o", " \\\r\n    --f"]
[1.681375, "o", "rom-beginn"]
[1.778239, "o", "ing \\\r\n    "]
[1.877473, "o", "--skip-mes"]
[1.975685, "o", "sage-on-er"]
[2.075257, "o", "ror \\\r\n    "]
[2.172982, "o", "--timeout-"]
[2.268925, "o", "ms 3000"]
[2.270572, "o", "\u001b[0m\r\n"]
[2.819898, "o", "[2024-01-23 01:16:57,410] INFO KafkaJsonSchemaDeserializerConfig values: \r\n\tauto.register.schemas = true\r\n\tbasic.auth.credentials.source = URL\r\n\tbasic.auth.user.info = [hidden]\r\n\tbearer.auth.cache.expiry.buffer.seconds = 300\r\n\tbearer.auth.client.id = null\r\n\tbearer.auth.client.secret = null\r\n\tbearer.auth.credentials.source = STATIC_TOKEN\r\n\tbearer.auth.custom.provider.class = null\r\n\tbearer.auth.identity.pool.id = null\r\n\tbearer.auth.issuer.endpoint.url = null\r\n\tbearer.auth.logical.cluster = null\r\n\tbearer.auth.scope = null\r\n\tbearer.auth.scope.claim.name = scope\r\n\tbearer.auth.sub.claim.name = sub\r\n\tbearer.auth.token = [hidden]\r\n\tcontext.name.strategy = class io.confluent.kafka.serializers.context.NullContextNameStrategy\r\n\thttp.connect.timeout.ms = 60000\r\n\thttp.read.timeout.ms = 60000\r\n\tid.compatibility.strict = true\r\n\tjson.fail.invalid.schema = true\r\n\tjson.fail.unknown.properties = true\r\n\tjson.key.type = class java.lang.Object\r\n\tjson.value.type = class java.lang.Object\r\n\tkey.subject.name.strategy = class io.conflu"]
[2.819926, "o", "ent.kafka.serializers.subject.TopicNameStrategy\r\n\tlatest.cache.size = 1000\r\n\tlatest.cache.ttl.sec = -1\r\n\tlatest.compatibility.strict = true\r\n\tmax.schemas.per.subject = 1000\r\n\tnormalize.schemas = false\r\n\tproxy.host = \r\n\tproxy.port = -1\r\n\trule.actions = []\r\n\trule.executors = []\r\n\trule.service.loader.enable = true\r\n\tschema.format = null\r\n\tschema.reflection = false\r\n\tschema.registry.basic.auth.user.info = [hidden]\r\n\tschema.registry.ssl.cipher.suites = null\r\n\tschema.registry.ssl.enabled.protocols = [TLSv1.2, TLSv1.3]\r\n\tschema.registry.ssl.endpoint.identification.algorithm = https\r\n\tschema.registry.ssl.engine.factory.class = null\r\n\tschema.registry.ssl.key.password = null\r\n\tschema.registry.ssl.keymanager.algorithm = SunX509\r\n\tschema.registry.ssl.keystore.certificate.chain = null\r\n\tschema.registry.ssl.keystore.key = null\r\n\tschema.registry.ssl.keystore.location = null\r\n\tschema.registry.ssl.keystore.password = null\r\n\tschema.registry.ssl.keystore.type = JKS\r\n\tschema.registry.ssl.protocol = TLSv1.3\r\n\tschema.registry.ssl."]
[2.820009, "o", "provider = null\r\n\tschema.registry.ssl.secure.random.implementation = null\r\n\tschema.registry.ssl.trustmanager.algorithm = PKIX\r\n\tschema.registry.ssl.truststore.certificates = null\r\n\tschema.registry.ssl.truststore.location = null\r\n\tschema.registry.ssl.truststore.password = null\r\n\tschema.registry.ssl.truststore.type = JKS\r\n\tschema.registry.url = [http://localhost:8081]\r\n\ttype.property = javaType\r\n\tuse.latest.version = false\r\n\tuse.latest.with.metadata = null\r\n\tuse.schema.id = -1\r\n\tvalue.subject.name.strategy = class io.confluent.kafka.serializers.subject.TopicNameStrategy\r\n (io.confluent.kafka.serializers.json.KafkaJsonSchemaDeserializerConfig:376)\r\n"]
[3.477395, "o", "[2024-01-23 01:16:58,067] ERROR Error processing message, skipping this message:  (kafka.tools.ConsoleConsumer$:44)\r\norg.apache.kafka.common.errors.SerializationException: Error deserializing JSON message for id 1\r\n\tat io.confluent.kafka.serializers.json.AbstractKafkaJsonSchemaDeserializer.deserialize(AbstractKafkaJsonSchemaDeserializer.java:236)\r\n\tat io.confluent.kafka.formatter.json.JsonSchemaMessageFormatter$JsonSchemaMessageDeserializer.deserialize(JsonSchemaMessageFormatter.java:135)\r\n\tat io.confluent.kafka.formatter.json.JsonSchemaMessageFormatter$JsonSchemaMessageDeserializer.deserialize(JsonSchemaMessageFormatter.java:101)\r\n\tat io.confluent.kafka.formatter.json.JsonSchemaMessageFormatter.writeTo(JsonSchemaMessageFormatter.java:92)\r\n\tat io.confluent.kafka.formatter.SchemaMessageFormatter.writeTo(SchemaMessageFormatter.java:266)\r\n\tat kafka.tools.ConsoleConsumer$.process(ConsoleConsumer.scala:116)\r\n\tat kafka.tools.ConsoleConsumer$.run(ConsoleConsumer.scala:76)\r\n\tat kafka.tools.ConsoleConsumer$.main(Conso"]
[3.477503, "o", "leConsumer.scala:53)\r\n\tat kafka.tools.ConsoleConsumer.main(ConsoleConsumer.scala)\r\nCaused by: org.apache.kafka.common.errors.SerializationException: JSON {\"name\":\"D\",\"age\":17,\"email\":\"bad email\",\"address\":{\"street\":\"a way too lond adress that will not fit in your database\",\"city\":\"\"},\"hobbies\":[\"reading\"],\"friends\":[{\"name\":\"Tom\",\"age\":17},{\"name\":\"Emma\",\"age\":18}]} does not match schema {\"$id\":\"https://example.com/user-schema.json\",\"$schema\":\"https://json-schema.org/draft/2020-12/schema\",\"type\":\"object\",\"properties\":{\"name\":{\"type\":\"string\",\"minLength\":3,\"maxLength\":50},\"age\":{\"type\":\"integer\",\"minimum\":0,\"maximum\":120},\"email\":{\"type\":\"string\",\"format\":\"email\"},\"address\":{\"type\":\"object\",\"properties\":{\"street\":{\"type\":\"string\",\"minLength\":5,\"maxLength\":15},\"city\":{\"type\":\"string\",\"minLength\":2,\"maxLength\":50}}},\"hobbies\":{\"type\":\"array\",\"items\":{\"type\":\"string\"},\"minItems\":2}}}\r\n\tat io.confluent.kafka.serializers.json.AbstractKafkaJsonSchemaDeserializer.deserialize(AbstractKafkaJsonSchemaDeserializer.java:1"]
[3.477516, "o", "83)\r\n\t... 8 more\r\nCaused by: org.everit.json.schema.ValidationException: #: 5 schema violations found\r\n\tat org.everit.json.schema.ValidationException.copy(ValidationException.java:486)\r\n\tat org.everit.json.schema.DefaultValidator.performValidation(Validator.java:76)\r\n\tat org.everit.json.schema.Schema.validate(Schema.java:152)\r\n\tat io.confluent.kafka.schemaregistry.json.JsonSchema.validate(JsonSchema.java:441)\r\n\tat io.confluent.kafka.schemaregistry.json.JsonSchema.validate(JsonSchema.java:409)\r\n\tat io.confluent.kafka.serializers.json.AbstractKafkaJsonSchemaDeserializer.deserialize(AbstractKafkaJsonSchemaDeserializer.java:178)\r\n\t... 8 more\r\n"]
[3.477666, "o", "[2024-01-23 01:16:58,067] ERROR Error processing message, skipping this message:  (kafka.tools.ConsoleConsumer$:44)\r\norg.apache.kafka.common.errors.SerializationException: Error deserializing JSON message for id 1\r\n\tat io.confluent.kafka.serializers.json.AbstractKafkaJsonSchemaDeserializer.deserialize(AbstractKafkaJsonSchemaDeserializer.java:236)\r\n\tat io.confluent.kafka.formatter.json.JsonSchemaMessageFormatter$JsonSchemaMessageDeserializer.deserialize(JsonSchemaMessageFormatter.java:135)\r\n\tat io.confluent.kafka.formatter.json.JsonSchemaMessageFormatter$JsonSchemaMessageDeserializer.deserialize(JsonSchemaMessageFormatter.java:101)\r\n\tat io.confluent.kafka.formatter.json.JsonSchemaMessageFormatter.writeTo(JsonSchemaMessageFormatter.java:92)\r\n\tat io.confluent.kafka.formatter.SchemaMessageFormatter.writeTo(SchemaMessageFormatter.java:266)\r\n\tat kafka.tools.ConsoleConsumer$.process(ConsoleConsumer.scala:116)\r\n\tat kafka.tools.ConsoleConsumer$.run(ConsoleConsumer.scala:76)\r\n\tat kafka.tools.ConsoleConsumer$.main(Conso"]
[3.477683, "o", "leConsumer.scala:53)\r\n\tat kafka.tools.ConsoleConsumer.main(ConsoleConsumer.scala)\r\nCaused by: org.apache.kafka.common.errors.SerializationException: JSON {\"name\":\"D\",\"age\":17,\"email\":\"bad email\",\"address\":{\"street\":\"a way too lond adress that will not fit in your database\",\"city\":\"\"},\"hobbies\":[\"reading\"],\"friends\":[{\"name\":\"Tom\",\"age\":17},{\"name\":\"Emma\",\"age\":18}]} does not match schema {\"$id\":\"https://example.com/user-schema.json\",\"$schema\":\"https://json-schema.org/draft/2020-12/schema\",\"type\":\"object\",\"properties\":{\"name\":{\"type\":\"string\",\"minLength\":3,\"maxLength\":50},\"age\":{\"type\":\"integer\",\"minimum\":0,\"maximum\":120},\"email\":{\"type\":\"string\",\"format\":\"email\"},\"address\":{\"type\":\"object\",\"properties\":{\"street\":{\"type\":\"string\",\"minLength\":5,\"maxLength\":15},\"city\":{\"type\":\"string\",\"minLength\":2,\"maxLength\":50}}},\"hobbies\":{\"type\":\"array\",\"items\":{\"type\":\"string\"},\"minItems\":2}}}\r\n\tat io.confluent.kafka.serializers.json.AbstractKafkaJsonSchemaDeserializer.deserialize(AbstractKafkaJsonSchemaDeserializer.java:1"]
[3.477694, "o", "83)\r\n\t... 8 more\r\nCaused by: org.everit.json.schema.ValidationException: #: 5 schema violations found\r\n\tat org.everit.json.schema.ValidationException.copy(ValidationException.java:486)\r\n\tat org.everit.json.schema.DefaultValidator.performValidation(Validator.java:76)\r\n\tat org.everit.json.schema.Schema.validate(Schema.java:152)\r\n\tat io.confluent.kafka.schemaregistry.json.JsonSchema.validate(JsonSchema.java:441)\r\n\tat io.confluent.kafka.schemaregistry.json.JsonSchema.validate(JsonSchema.java:409)\r\n\tat io.confluent.kafka.serializers.json.AbstractKafkaJsonSchemaDeserializer.deserialize(AbstractKafkaJsonSchemaDeserializer.java:178)\r\n\t... 8 more\r\n"]
[3.478521, "o", "{\"name\":\"Doe\",\"age\":17,\"email\":\"john.doe@example.com\",\"address\":{\"street\":\"123 Main St\",\"city\":\"Anytown\"},\"hobbies\":[\"reading\",\"cycling\"],\"friends\":[{\"name\":\"Tom\",\"age\":9},{\"name\":\"Emma\",\"age\":10}]}\r\n"]
[6.482617, "o", "[2024-01-23 01:17:01,073] ERROR Error processing message, terminating consumer process:  (kafka.tools.ConsoleConsumer$:44)\r\norg.apache.kafka.common.errors.TimeoutException\r\n[2024-01-23 01:17:01,073] ERROR Error processing message, terminating consumer process:  (kafka.tools.ConsoleConsumer$:44)\r\norg.apache.kafka.common.errors.TimeoutException\r\n"]
[6.505263, "o", "Processed a total of 2 messages\r\n"]
